debug/custom3d.html

```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Mapbox GL JS debug page</title>
+    <meta charset='utf-8'>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
+    <link rel='stylesheet' href='/dist/mapbox-gl.css' />
+    <style>
+        body { margin: 0; padding: 0; }
+        html, body, #map { height: 100%; }
+    </style>
+</head>
+
+<body>
+<div id='map'></div>
+
+<script src='/dist/mapbox-gl-dev.js'></script>
+<script src='/debug/access_token_generated.js'></script>
+<script>
+
+
+var map = window.map = new mapboxgl.Map({
+    container: 'map',
+    zoom: 1.69,
+    center: [-9.4, -26.8],
+    bearing: 131,
+    pitch: 60,
+    style: 'mapbox://styles/mapbox/light-v9',
+    hash: true
+});
+
+class Custom {
+    constructor() {
+        this.id = 'mycustomlayer';
+        this.type = 'custom';
+        this.renderingMode = '3d';
+    }
+
+    onAdd(map, gl) {
+
+        const vertexSource = `
+
+        attribute vec3 aPos;
+        uniform mat4 uMatrix;
+
+        void main() {
+            gl_Position = uMatrix * vec4(aPos, 1.0);
+        }
+        `;
+
+        const fragmentSource = `
+        void main() {
+            gl_FragColor = vec4(1.0, 0.0, 0.0, 1.0);
+        }
+        `;
+
+        const vertexShader = gl.createShader(gl.VERTEX_SHADER);
+        gl.shaderSource(vertexShader, vertexSource);
+        gl.compileShader(vertexShader);
+        const fragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+        gl.shaderSource(fragmentShader, fragmentSource);
+        gl.compileShader(fragmentShader);
+
+        this.program = gl.createProgram();
+        gl.attachShader(this.program, vertexShader);
+        gl.attachShader(this.program, fragmentShader);
+        gl.linkProgram(this.program);
+        gl.validateProgram(this.program);
+
+        this.program.aPos = gl.getAttribLocation(this.program, "aPos");
+        this.program.uMatrix = gl.getUniformLocation(this.program, "uMatrix");
+
+        const x = 0.5;
+        const y = 0.5;
+        const z = 0.125;
+        const d = 0.125;
+
+        const vertexArray = new window.Float32Array([
+            x, y, 0,
+            x + d, y, 0,
+            x, y + d, z,
+            x + d, y + d, z,
+            x, y + d + d, 0,
+            x + d, y + d + d, 0]);
+        const indexArray = new window.Uint16Array([
```

Comment:
Nit: why do we access those with `window`? If linter complains, this probably needs a fix on the lint rules side.